oauth2-server-snap/lib/Network/OAuth2/Server/Snap.hs

```diff
@@ -64,6 +104,15 @@ createGrant
     :: Handler b (OAuth2 IO b) TokenGrant
 createGrant = return aGrant
 
+-- | Create an access token and send it to the client.
+createAndServeToken
```

Comment:
We need a `GrantRequest` type for this to take as a param, encompassing type, scope, etc.